### PR TITLE
Add Java tracer compatibility matrix and new download links

### PR DESCRIPTION
**Pull request by PerfectSlayer**
### What does this PR do?

This PR describes the JVM compatibility for Java tracer.
Additionally, it introduces the new download link for major versions.

### Motivation

We need to define clear statement about supported JVM for the current and coming versions of the Java tracer.

### Additional Notes

@bantonsson Can you review the content to check it matches what you expected?
Compared to the original document, I added mentions to Oracle JDK and OpenJDK, and also the non-support rule for early access builds.

---

### Reviewer checklist
- [x] Review the changed files.
- [x] Review the URLs listed in the [Preview](#preview) section.
- [x] Check images for PII
- [x] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/trace_collection/compatibility/java.md [hunk 1]
```diff
@@ -19,7 +19,28 @@ further_reading:
...
 
-Datadog officially supports the Java JRE 1.7 and higher of both Oracle JDK and OpenJDK. Datadog does not officially support any early-access versions of Java.
+### Supported JVM runtimes
+
+The Java Tracer supports automatic instrumentation of the following Oracle JDK and OpenJDK JVM runtimes.
+
+| JVM versions | Operating Systems                                                 | Support level         | Tracer version |
+| -------------| ----------------------------------------------------------------- | --------------------- | -------------- |
+| 7 to 17      | Windows (x86, x86-64)<br>Linux (x86, x86-64)<br>Mac (x86, x86-64) | [GA](#support-ga)     | latest         |
+| 7 to 17      | Linux (arm64)<br>Mac (arm64)                                      | [Beta](#support-beta) | latest         |
```
Comment (bantonsson) [inline content/en/tracing/trace_collection/compatibility/java.md:29]: I think we should add 18,19 as Beta

**Review by bantonsson — APPROVED** [review]
Looks good. Great to keep the e-a disclaimer.

File: content/en/tracing/trace_collection/compatibility/java.md (hunk 1 above)
Comment (PerfectSlayer) [reply content/en/tracing/trace_collection/compatibility/java.md]: Done! 👌

File: content/en/tracing/trace_collection/compatibility/java.md [hunk 2]
```diff
@@ -19,7 +19,29 @@ further_reading:
...
 The Java Datadog Trace library is open source - view the [GitHub repository][1] for more information.
 
-Datadog officially supports the Java JRE 1.7 and higher of both Oracle JDK and OpenJDK. Datadog does not officially support any early-access versions of Java.
+### Supported JVM runtimes
+
+The Java Tracer supports automatic instrumentation of the following Oracle JDK and OpenJDK JVM runtimes.
+
+| JVM versions | Operating Systems                                                             | Support level             | Tracer version |
+| -------------| ----------------------------------------------------------------------------- | ------------------------- | -------------- |
+| 18 to 19     | Windows (x86, x86-64)<br>Linux (x86, x86-, arm64)<br>Mac (x86, x86-64, arm64) | [Beta](#support-beta)     | latest         |
```
Comment (maycmlee) [inline content/en/tracing/trace_collection/compatibility/java.md:28]:
```suggestion
| 18 to 19     | Windows (x86, x86-64)<br>Linux (x86, x86-, arm64)<br>Mac (x86, x86-64, arm64) | [Beta](#support-beta)     | Latest         |
```

File: content/en/tracing/trace_collection/compatibility/java.md [hunk 3]
```diff
@@ -19,7 +19,29 @@ further_reading:
...
+| 7 to 17      | Windows (x86, x86-64)<br>Linux (x86, x86-64)<br>Mac (x86, x86-64)             | [GA](#support-ga)         | latest         |
```
Comment (maycmlee) [inline content/en/tracing/trace_collection/compatibility/java.md:29]:
```suggestion
| 7 to 17      | Windows (x86, x86-64)<br>Linux (x86, x86-64)<br>Mac (x86, x86-64)             | [GA](#support-ga)         | Latest         |
```

File: content/en/tracing/trace_collection/compatibility/java.md [hunk 4]
```diff
@@ -19,7 +19,29 @@ further_reading:
...
+| 7 to 17      | Linux (arm64)<br>Mac (arm64)                                                  | [Beta](#support-beta)     | latest         |
```
Comment (maycmlee) [inline content/en/tracing/trace_collection/compatibility/java.md:30]:
```suggestion
| 7 to 17      | Linux (arm64)<br>Mac (arm64)                                                  | [Beta](#support-beta)     | Latest         |
```

File: content/en/tracing/trace_collection/compatibility/java.md [hunk 5]
```diff
@@ -19,7 +19,29 @@ further_reading:
 
 The Java Datadog Trace library is open source - view the [GitHub repository][1] for more information.
 
-Datadog officially supports the Java JRE 1.7 and higher of both Oracle JDK and OpenJDK. Datadog does not officially support any early-access versions of Java.
+### Supported JVM runtimes
+
+The Java Tracer supports automatic instrumentation of the following Oracle JDK and OpenJDK JVM runtimes.
```
Comment (maycmlee) [inline content/en/tracing/trace_collection/compatibility/java.md:24]:
```suggestion
The Java Tracer supports automatic instrumentation for the following Oracle JDK and OpenJDK JVM runtimes.
```

File: content/en/tracing/trace_collection/dd_libraries/java.md [hunk 6]
```diff
@@ -111,7 +111,10 @@ After the agent is installed, to begin tracing your applications:
    ```shell
    wget -O dd-java-agent.jar https://dtdg.co/latest-java-tracer
    ```
-   To access a specific tracer version, visit Datadog's [Maven repository][3].
+   
+   **Note:** To download a specific major version, use the `https://dtdg.co/java-tracer-vX` link instead, where `vX` is the desired vesion
```
Comment (maycmlee) [inline content/en/tracing/trace_collection/dd_libraries/java.md:115]:
```suggestion
   **Note:** To download a specific major version, use the `https://dtdg.co/java-tracer-vX` link instead, where `vX` is the desired version.
```

File: content/en/tracing/trace_collection/dd_libraries/java.md [hunk 7]
```diff
@@ -111,7 +111,10 @@ After the agent is installed, to begin tracing your applications:
...
+   (`https://dtdg.co/java-tracer-v0` for latest version 0 for example).
```
Comment (maycmlee) [inline content/en/tracing/trace_collection/dd_libraries/java.md:116]:
```suggestion
   For example, use `https://dtdg.co/java-tracer-v0` for the latest version 0.
```

**Comment by PerfectSlayer** [comment]
Thanks @maycmlee for your review! 🙏 

About your suggestions for the _levels of support table_, it comes from the [.net page](https://docs.datadoghq.com/tracing/trace_collection/compatibility/dotnet-framework#levels-of-support).
Would you like me to update this page too with your comments, or only stick to my java part?

**Review by maycmlee — APPROVED** [review]
Hey @PerfectSlayer! Thanks for letting me know it was from the .NET page. Looks like you committed my changes, so let's just leave it. Is this good to be merged?

**Comment by PerfectSlayer** [comment]
It is! 👌

**Pull request merged**